supermarq-benchmarks/examples/qre-challenge/Lessons-Learned.md

```diff
@@ -0,0 +1,106 @@
+
+_Trapped Ions_
+
+Trapped-ion qubits leverage ions confined in electromagnetic fields as qubits, with quantum operations performed via laser pulses or microwave fields. This platform is known for high-fidelity operations and long coherence times, making it a robust option for quantum computing. The precise control over qubit interactions and the ability to implement complex algorithms are key strengths. However, scalability remains a challenge due to the physical limitations of ion trapping and the complexity of managing large ion chains.
+
+_Optical Tweezer_
+
+Optical tweezer qubits involve neutral atoms individually trapped by tightly focused laser beams. These atoms can be arranged in customizable patterns, allowing for flexible and highly connected qubit geometries. Optical tweezers are promising for large-scale quantum computing due to their natural scalability and the ease of reconfiguring qubit layouts. Despite these advantages, maintaining long-term coherence and achieving consistent high-fidelity operations across many qubits are significant challenges that need to be addressed.
+
+_Comparison and Implications for QRE_
+
+The choice of qubit architecture—synthetic versus neutral atom—has significant implications for QRE. Superconducting qubits often require careful consideration of connectivity due to their fixed physical layout, while neutral atoms offer more flexibility in qubit arrangement but face challenges in operation fidelity. A comprehensive QRE tool must account for these differences in architecture, adjusting resource estimations based on the type of qubit used, its coherence properties, and the specific operational constraints associated with each platform.
+
+### **Requirements**
+
+_Functional Requirements_
+
+- **Movement Analysis:** The tool must evaluate the impact of qubit movement within the architecture.
+
+- **Atom Reloading:** Consideration of the frequency and impact of atom reloading within quantum operations.
+
+- **Code Teleportation:** Ability to model and assess the implications of code teleportation on qubit resources.
+
+- **Connectivity:** Analysis of qubit connectivity and its effect on the overall architecture.
+
+- **Degree of Parallelization:** Estimation of the impact of parallel operations on qubit usage and efficiency.
+
+- **Custom Assumptions:** Flexibility to incorporate various architectural assumptions and parameters.
+
+- **Universal Gateset Configuration:** The tool must support the ability to implement a universal gateset without relying on magic state distillation, allowing for more versatile quantum operations.
+
+- **Flexible QEC Code Switching:** Users should be able to switch between different quantum error correction codes, such as 3D codes, subsystem codes, concatenated codes, and piece-able fault tolerance, providing a broader range of options for optimizing quantum systems.
```

Comment:
these two points need to be clarified/disentangled a bit imo. "code switching" refers to something specific, which doesn't encompass everything in that bullet - e.g. "piece-able fault tolerance" is a way to implement a complete gateset (and so seems like it belongs in the previous point as an alternative to magic state distillation?)

would something like this be close to what you're going for?
1. it must consider a variety of techniques for implementing a universal gateset, including:
    * magic state distillation
    * code switching
    * piece-able fault tolerance
    * lattice surgery
    * ...
2. it must support arbitrary (or at least a wide variety of) QEC codes, ...